python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -239,15 +280,29 @@ def __init__(
 
         # Get all available layers (including keras layers)
         self.available_layers = get_layer_subclasses()
-        model_graph = self.define_architecture(model_config)
-        self.execution_order: List[LayerNode] = model_graph.topological_sort()
+
+        # Visualize with pygraphviz if available
+        self.viz_path = None
+        if pgv:
+            # Get the first file handler for the logger
+            file_handlers = [handler for handler in self.file_io.logger.handlers if isinstance(handler, FileHandler)]
+            logging_dir = Path(file_handlers[0].baseFilename).parent if file_handlers else Path(self.DEFAULT_VIZ_SAVE_PATH)
```

Comment:
@arvindsrikantan I remember leaving a comment about not fetching the folder from the logger. Why can't we move this logging higher up the call to `RelevanceScorer` perhaps?

Reply:
You could have a `draw(...)` method for both `DNN` and `AutoDagNetwork` and leave it unimplemented for DNN

Reply:
Comment got lost in the refactoring I guess. IIRC, I was talking about how the visualization was specific to this architecture type, but I now see what you meant

Reply:
Okay, let me know if you are addressing this. Otherwise I can approve this PR. Everything else seems to be handled to me.